src/elements/toniq-dropdown/toniq-dropdown.element.ts

```diff
@@ -0,0 +1,188 @@
+import {assign, css, defineElementEvent, html} from 'element-vir';
+import {ReactiveController, ReactiveControllerHost} from 'lit';
+import {ChevronDown24Icon} from '../../icons';
+import {applyBackgroundAndForeground, toniqColors} from '../../styles/colors';
+import {defineToniqElement} from '../define-toniq-element';
+import {ToniqIcon} from '../toniq-icon/toniq-icon.element';
+
+export interface ToniqDropdownSelect {
+    value: string | number | undefined;
+    label: string | undefined;
+}
+
+export class ToniqDropdownController implements ReactiveController {
+    host: ReactiveControllerHost;
+    dropDownEl: HTMLElement;
+
+    constructor(host: ReactiveControllerHost, dropDownEl: HTMLElement) {
+        (this.host = host).addController(this);
+        this.dropDownEl = dropDownEl;
+    }
+
+    clickOutside(event: Event) {
+        const withinBoundaries = event.composedPath().includes(this.dropDownEl);
+        if (!withinBoundaries && this.dropDownEl && this.dropDownEl.classList.contains('open')) {
+            this.dropDownEl.classList.remove('open');
+        }
+    }
+
+    hostConnected() {
+        window.addEventListener('click', this.clickOutside.bind(this));
+    }
+
+    hostDisconnected() {
+        window.removeEventListener('click', this.clickOutside.bind(this));
+    }
+}
+
+export const ToniqDropdown = defineToniqElement({
+    tagName: 'toniq-dropdown',
+    props: {
+        list: [] as undefined | ToniqDropdownSelect[],
+        select: {} as undefined | ToniqDropdownSelect,
+    },
+    events: {
+        selectChange: defineElementEvent<object>(),
+    },
+    styles: css`
+        :host {
+            font-family: var(--toniq-font, 'Rubik');
+            font-style: normal;
+            font-weight: 700;
+            font-size: 16px;
+            line-height: 24px;
+        }
+
+        .dropdown {
+            width: 288px;
+            height: 48px;
+            position: relative;
+        }
+
+        toniq-icon {
+            -moz-transition: all 0.2s linear;
+            -webkit-transition: all 0.2s linear;
+            transition: all 0.2s linear;
+        }
+
+        .dropdown.open toniq-icon {
+            -moz-transform: rotate(180deg);
+            -webkit-transform: rotate(180deg);
+            transform: rotate(180deg);
+        }
+
+        .dropdown.open .select-options {
+            display: grid;
+        }
+
+        .select {
+            display: flex;
+            justify-content: space-between;
+            border-radius: 8px;
```

Comment:
When the dropdown is open, the bottom border radii should be removed to prevent weird gaps:

![Screen Shot 2022-06-03 at 09 06 12](https://user-images.githubusercontent.com/1205860/171881495-7fd0e078-7bdc-4f9e-836a-b716f6dbc5a7.png)